components/raftstore/src/store/peer.rs

```diff
@@ -1172,8 +1256,6 @@ impl Peer {
         // could get a newer value, and after that, the leader may read a stale value,
         // which violates linearizability.
         self.get_store().applied_index() >= read_index
-            && !self.is_splitting()
-            && !self.is_merging()
```

Comment:
This should not be removed!

Reply:
They are useless. `is_splitting` and `is_merging` should only be used on leader for its lease.
Actually, the `last_committed_split_idx` and `last_committed_prepare_merge_idx` are maintained only on leader all the time.

Reply:
The `pending_merge_state` in `is_merging` is important for follower read. I recover this code.